cpp/src/groupby/hash/groupby.cu

```diff
@@ -65,6 +65,13 @@ namespace detail {
 namespace hash {
 namespace {
 
+using map_type = concurrent_unordered_map<
```

Comment:
If possible, please add comment clarifying why we are using this map type. I've seen that we use `static_map` somewhere else, and here `concurrent_unordered_map`.

Reply:
Replacing `groupby::hash`'s `concurrent_unordered_map` with `cuco::static_map` will be the following PR right after. Eventually, we will completely get rid of `concurrent_unordered_map` in cudf (https://github.com/rapidsai/cudf/issues/10401).